### DISCUSSION TITLE
Lesson 7, Part 9 (HelperConfig) - testPriceFeedVersionIsAccurate fails for me but not for Patrick

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#505] [General]
Never mind, I figured out what I did wrong...I trusted the address suggested by Github AI and it wasn't the correct address for the ETH/USD fee for Sepolia per Chainlink's docs...oops. But I don't know how to delete this...only how to edit it. No need to answer though.

I just did the exercise where you set up a HelperConfig file to set up the network configuration for multiple different chains. It compiled successfully and then ran the 3 tests (testMinimumDollarIsFive, testOwnerIsMsgSender, and testPriceFeedVersionIsAccurate) but my code failed the testPriceFeedVersionIsAccurate test whereas Patrick's did not. My screencap is below.

I did console.log(version) and used -vvvv to get as much info as I could - you can see what I got below - and, as best I can tell, it seems like the issue is because fundMe is now being deployed through the DeployFundMe.s.sol script (rather than being an instance of the FundMe.sol contract that had the getVersion function in it). But I'm curious why the test passed for Patrick but not for me.

I'm also wondering why there wasn't an issue with fundMe.MINIMUM_USD or fundMe.i_owner ....obviously those are variables rather than functions so maybe different.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/138697105/386259e9-1be7-4b17-a502-46fdc43234b7)

**Comment by 0xTagger** [comment]
Hi @0xwtfwtf - How did you solve this? I am getting the same error, but for me the Sepolia ETH/USD address is the correct one in the HelperConfig.s.sol file.
<img width="1378" alt="Screenshot 2023-08-07 at 10 44 59 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/0087a175-9025-46cb-be96-62627bb5f471">

**Comment by theirrationalone** [comment]
Please update version number to 4 in... By default it's 0.

```
test/mocks/MockV3Aggregator.m.sol
```
Or try to test version == 0.

**Reply by 0xTagger** [reply]
I'm sorry, but I'm confused by what you are referring to. In the screenshot I posted, the version number is 4 inside FundMeTest.t.sol.

Am I missing something?

**Reply by theirrationalone** [reply]
Please also update that inside MockV3Aggregator.m.sol, file that's is test/mocks directory

**Reply by 0xTagger** [reply]
I have not had to use that file yet. I don't think I'm that far ahead in Lesson 7 as I have not used or seen MockV3Aggregator.m.sol in the videos.

I am on Lesson 7 - Part 9 - Refactoring II Helper Config @ 9:54 minutes

**Reply by theirrationalone** [reply]
So limit yourself and make assert like this.. 

```solidity
assertEq(version, 0);
```

and do console.log(version); before assertEq

**Reply by 0xTagger** [reply]
I am still receiving the same error after trying the solution you stated.
<img width="1111" alt="Screenshot 2023-08-09 at 8 53 46 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/dcc453b2-f611-43a4-93e5-d429861b6ceb">

**Reply by theirrationalone** [reply]
Actually you're missing an excerpt of patrick's video Or might you're trying without finishing the lesson. 

You haven't added that `MockV3Aggregator`

Please complete the lesson first.

**Reply by theirrationalone** [reply]
Start Watching 2nd [video](https://youtu.be/sas02qSFZ74) from timestamp: 55:50

**Reply by 0xTagger** [reply]
I wasn't on the MockV3Aggregator part yet, but I figured it out. It was because I added an additional 1 to the Sepolia Blockchain ID in the HelperConfig.s.sol file... 111551111 instead of 11155111.

Thank you for your efforts in helping me solve this - it is much appreciated!

**Reply by theirrationalone** [reply]
I'm now surprised, if u're not using mockV3Aggregator yet and you're testing it locally then... 

How that error got mitigated??? 

`getVersion` is available only in priceFeed(instance).

If it was that excess 1 in 11155111(sepolia chainid) then still it was not the problem because every time when tests run then that check block.chainid == 111551111 will get false and control will get shifted to anvil(configs). 

So thus far, The result might still came same.

What's the heck?? 

This is definately you're missing `MockV3Aggregator` and test will get failed until you add mocks. And you said "I wasn't on the MockV3Aggregator part yet" So please implement mocks And then try again.

**Reply by hammadshub** [reply]
In HelperConfig.s.sol , make sure your using exact like this in mainnet else if
```solidity
constructor() {
        if (block.chainid == 11155111) {
            activeNetworkConfig = getSepoliaEthConfig();
        } else if (block.chainid == 1) {
            activeNetworkConfig = getMainnetEthConfig();
        } else {
            activeNetworkConfig = getAnvilEthConfig();
        }
    }
```
```solidity
else if (block.chainid == 1) {
            activeNetworkConfig = getMainnetEthConfig();
```
